src/interface/viewport.c

```diff
@@ -1113,6 +1113,72 @@ int sub_98198C(int image_id, sint8 x_offset, sint8 y_offset, sint16 length_x, si
 }
 
 /**
+ * rct2: 0x006D42F0
+ */
+static void vehicle_visual_mini_golf_player(int x, int imageDirection, int y, int z, rct_vehicle *vehicle, int rct2VehiclePtrFormat) {
+	RCT2_CALLPROC_X(0x006D42F0, x, imageDirection, y, z, (int) vehicle, rct2VehiclePtrFormat, 0);
+	//return;
+
+	if (vehicle->num_peeps == 0) {
+		return;
+	}
+
+	rct_drawpixelinfo *edi = RCT2_GLOBAL(0x0140E9A8, rct_drawpixelinfo*);
+	if (edi->zoom_level >= 2) {
+		return;
+	}
+
+	if (RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_VIEWPORT_FLAGS, uint16) & VIEWPORT_FLAG_INVISIBLE_PEEPS) {
+		return;
+	}
+
+	rct_ride_entry *rideType = get_ride_entry(get_ride(vehicle->ride)->subtype);
+	rct_sprite *sprite = &g_sprite_list[vehicle->peep[0]];
+
+	uint32 eax = RCT2_ADDRESS(0x008B8F74, uint32)[vehicle->var_D4];
+	uint32 ebx = (RCT2_GLOBAL(eax + vehicle->var_C5, uint8) << 2) + (imageDirection >> 3);
+
+	uint32 image_id = rideType->vehicles[0].base_image_id + 1 + ebx;
+
+	uint32 peep_palette = sprite->peep.tshirt_colour << 19 | sprite->peep.trousers_colour << 24 | 0x0A0000000;
+
+	RCT2_GLOBAL(0x9DEA52, uint16) = 0;
+	RCT2_GLOBAL(0x9DEA54, uint16) = 0;
+	RCT2_GLOBAL(0x9DEA56, uint16) = z + 5;
```

Comment:
these are used as arguments to sub_98197C so they are not needed, and you forgot to add 5 to z on that, same with ball.